back/src/common/redis/webhooksettings.ts

```diff
@@ -59,31 +57,50 @@ export async function clearWebhookSetting(cursor = 0): Promise<void> {
   }
 }
 
-export async function delWebhookSetting(orgId: string): Promise<void> {
-  const key = genWebhookKey(orgId);
-  await redisClient.del(key);
+export async function delWebhookSetting(
+  webhookSetting: WebhookSetting
+): Promise<void> {
+  const key = genWebhookKey(webhookSetting.orgId);
+  await redisClient.srem(key, smember(webhookSetting));
 }
+// How many failed webhook for a given orgId before deactivation
 const WEBHOOK_FAIL_ACCEPTED = parseInt(
   process.env.WEBHOOK_FAIL_ACCEPTED || "5",
   10
-); // how many failed webhook ofor a given orgId before deactivation
+);
+// How long after the last fail the counter is reset
 const WEBHOOK_FAIL_RESET_DELAY = parseInt(
   process.env.WEBHOOK_FAIL_RESET_DELAY || "600",
   10
-); // how long after the last fail the counter is reset
+);
 
-const genWebhookFailKey = (orgId: string): string =>
-  `${WEBHOOK_FAIL_CACHE_KEY}:${orgId}`;
+const genWebhookFailKey = (orgId: string, endpointUri: string): string =>
+  `${WEBHOOK_FAIL_CACHE_KEY}:${orgId}:${endpointUri}`;
 
-export async function handleWebhookFail(orgId: string): Promise<void> {
-  const key = genWebhookFailKey(orgId);
+export async function handleWebhookFail(
+  orgId: string,
+  endpointUri: string
+): Promise<void> {
+  const key = genWebhookFailKey(orgId, endpointUri);
   const failCount = await redisClient.get(key);
   if (failCount && parseInt(failCount, 10) >= WEBHOOK_FAIL_ACCEPTED) {
-    await prisma.webhookSetting.update({
-      where: { orgId },
-      data: { activated: false }
+    const wehbookSetting = await prisma.webhookSetting.findFirst({
+      where: { endpointUri, orgId }
     });
-    await delWebhookSetting(orgId);
+
+    if (wehbookSetting) {
+      await prisma.webhookSetting.update({
+        where: {
+          WebhookSetting_orgId_endpointUri_unique_together: {
```

Comment:
C'est vraiment comme ça qu'on écrit une query sur un index composé unique? Je trouve ça chelou

Reply:
c'est bien ca, ils en parlent la: https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/working-with-composite-ids-and-constraints